### DISCUSSION TITLE
Ordered Indexes and Conditions

### DISCUSSION THREAD

**Discussion by theelderbeever** [piccolo-orm/piccolo#531] [Ideas]
I am not sure if SQLite supports them but, Postgres has support for Ordered Indexes and Conditional Indexes. They likely aren't common use case scenarios however, support for them would be nice.

### Ordered Indexes

```sql
CREATE INDEX sales_timestamp ON sales ("timestamp" DESC);

-- Needs to be extendable to the eventual composite index case
CREATE INDEX sales_timestamp_id ON sales ("index" DESC, "timestamp" DESC);
```

### Conditional Indexes

```sql
CREATE INDEX sales_price ON sales ("price") WHERE price > 9000;

CREATE INDEX sales_buyer ON sales ("buyer") WHERE buyer IS NOT NULL;
```

## Workaround

Simple workaround of using a raw migration at the moment.

**Comment by dantownsend** [comment]
Interesting - I haven't played around with this much, but it should be possible.
